api/src/test/java/org/openmrs/util/OpenmrsUtilUnitTest.java

```diff
@@ -137,4 +147,330 @@ public void join_shouldReturnListElementsJoinedBySeparatorIfGivenSeparatorIsNull
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInAbsoluteNumericRange_shouldReturnFalseIfValueIsBelowRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(10.34d);
+		concept.setLowCritical(6.67d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInAbsoluteNumericRange_shouldReturnTrueIfValueIsEqualToHiRangeLimit() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(4.34d);
+		concept.setLowCritical(3.67d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+	@Test
+	public void isInAbsoluteNumericRange_shouldReturnTrueIfValueIsEqualToLowRangeLimit() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(14.34d);
+		concept.setLowCritical(4.34d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfHiAbsoluteIsNull() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(null);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfLowAbsoluteIsNull() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setLowAbsolute(null);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfValueIsInRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(3.67d);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnFalseIfValueIsAboveRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(6.67d);
+
+		assertFalse(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnFalseIfValueIsBelowRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(6.67d);
+
+		assertFalse(OpenmrsUtil.isValidNumericValue(15.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfValueIsEqualToHiRangeLimit() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(4.34d);
+		concept.setLowCritical(3.67d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfValueIsEqualToLowRangeLimit() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(14.34d);
+		concept.setLowCritical(4.34d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+
+	@Test
+	public void isTestMode_shouldReturnTrueIfFunctionalTestModeIsEnabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", "true");
```

Comment:
can you please check with `System.getProperty` what value `FUNCTIONAL_TEST_MODE` is in without these two tests that set a value.

we should make sure that we restore the value to its previous state so we dont influence other tests in any way.

you could add a 
```java
@After
public clearSystemProperty() {
    System.clearProperty("FUNCTIONAL_TEST_MODE")
}
```

to the top of the test